sky/backends/backend.py

```diff
@@ -51,6 +56,14 @@ def provision(
             retry_until_up: bool = False) -> Optional[_ResourceHandleType]:
         if cluster_name is None:
             cluster_name = sky.backends.backend_utils.generate_cluster_name()
+        else:
+            prev_cluster_name = cluster_name
+            cluster_name = adjust_cluster_name(cluster_name)
```

Comment:
Only adjusting the cluster_name here may not be enough. For example, what if the user specify `sky launch -c test_myLoRA`, and would run `sky status test_myLoRA` later?

For discussion, I am wondering if it is possible that we make all the cluster name valid from the user side, and directly show the cluster name the user specified in the `sky status`, while we do the adjustment when we actually tries to launch on the cloud. Basically we only modify the `cluster_name_on_cloud`.

Reply:
During the team's discussion offline, the decision agreed upon is to allow for users to include `.`, `_`, and uppercases, on top of the previously allowed characters. The cluster shown for sky status and cluster name stored in the DB would be the same as the user input while the cloud cluster name could be different based on transformations of the newly included characters